### ISSUE TITLE
v3.0.5 - Android bugs.

### ISSUE THREAD

**Issue by aykutuludag** [gdelataillade/alarm#137]
![image](https://github.com/gdelataillade/alarm/assets/18382503/512e36c6-0bcc-44c6-b59d-97a05fe98030)

```
Fatal Exception: java.lang.RuntimeException: Unable to start service com.gdelataillade.alarm.alarm.AlarmService@c777b30 with Intent { cmp=com.appname/com.gdelataillade.alarm.alarm.AlarmService (has extras) }: android.app.ForegroundServiceStartNotAllowedException: Service.startForeground() not allowed due to mAllowStartForeground false: service com.appname/com.gdelataillade.alarm.alarm.AlarmService
       at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4963)
       at android.app.ActivityThread.-$$Nest$mhandleServiceArgs()
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2288)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loopOnce(Looper.java:211)
       at android.os.Looper.loop(Looper.java:300)
       at android.app.ActivityThread.main(ActivityThread.java:8410)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:559)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:954)
```
Other than this error, I couldn't detect any error so far on Android. Best regards.

**Note**: My suspect that this is only occurring when app is open alarm is triggered at that moment. Service couldn't start foreground, I guess.

**Comment by aykutuludag** [comment]
Suggestion: I found some part in your Android code.

```
if (fullScreen) {
            notificationBuilder.setFullScreenIntent(notificationPendingIntent, true)
 }
```

I guess this part opens notification automatically but in best scenario we give this choice to user either open notification or dismiss it. I believe that this part should be removed or should be optional. (ex: bool autoOpenNotification: true/false). Have a nice weekend.

**Comment by aykutuludag** [comment]
Suggestion: I changed your code a little bit for myself to show notification icon:

```
        val bm: Bitmap = BitmapFactory.decodeResource(context.getResources(), iconResId)

        notificationBuilder
            .setSmallIcon(iconResId)
            .setLargeIcon(bm)
            .setContentTitle(title)
            .setContentText(body)
            .setPriority(NotificationCompat.PRIORITY_MAX)
            .setCategory(NotificationCompat.CATEGORY_ALARM)
            .setAutoCancel(true)
            .setContentIntent(notificationPendingIntent)
            .setSound(null)
            .setVisibility(NotificationCompat.VISIBILITY_PUBLIC)
```

I added  **.setLargeIcon(bm)**, in this way users can see application icon inside notification.

**Comment by gdelataillade** [comment]
Hi @aykutuludag 

About the full screen notification, there's a parameter for this: `androidFullScreenIntent`, true by default. Have you tried to disable it ?

**Comment by gdelataillade** [comment]
About the notification large icon, thanks for sharing your code. I'll add it to the next release !

**Comment by aykutuludag** [comment]
> Hi @aykutuludag
> 
> About the full screen notification, there's a parameter for this: `androidFullScreenIntent`, true by default. Have you tried to disable it ?

You are right!  My mistake I couldn't see that variable. But even I removed that part still alarms and notifications are working :D I can set it up as false in next version, but my suggestion is that user can able to choose this from dart side. Most people doesn't know kotlin or swift :D Kind regards, have a nice day.

**Comment by aykutuludag** [comment]
> ![image](https://private-user-images.githubusercontent.com/18382503/298094320-512e36c6-0bcc-44c6-b59d-97a05fe98030.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU4MzczNDYsIm5iZiI6MTcwNTgzNzA0NiwicGF0aCI6Ii8xODM4MjUwMy8yOTgwOTQzMjAtNTEyZTM2YzYtMGJjYy00NGM2LWI1OWQtOTdhMDVmZTk4MDMwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTIxVDExMzcyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRiYjJlZmYzNWUyMzAxYjgxOTc4ODU4OTMxZDUzZDkyYzhmODQ5ODE0NWRhZjQ5ZTUwMzA1MzIzYWNkODdkY2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4MhH5TcGAzViBd5kEKq2sND1GMPCoyAUFFZmt_2_szs)
> 
> ```
> Fatal Exception: java.lang.RuntimeException: Unable to start service com.gdelataillade.alarm.alarm.AlarmService@c777b30 with Intent { cmp=com.appname/com.gdelataillade.alarm.alarm.AlarmService (has extras) }: android.app.ForegroundServiceStartNotAllowedException: Service.startForeground() not allowed due to mAllowStartForeground false: service com.appname/com.gdelataillade.alarm.alarm.AlarmService
>        at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4963)
>        at android.app.ActivityThread.-$$Nest$mhandleServiceArgs()
>        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2288)
>        at android.os.Handler.dispatchMessage(Handler.java:106)
>        at android.os.Looper.loopOnce(Looper.java:211)
>        at android.os.Looper.loop(Looper.java:300)
>        at android.app.ActivityThread.main(ActivityThread.java:8410)
>        at java.lang.reflect.Method.invoke(Method.java)
>        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:559)
>        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:954)
> ```
> 
> Other than this error, I couldn't detect any error so far on Android. Best regards.
> 
> **Note**: My suspect that this is only occurring when app is open alarm is triggered at that moment. Service couldn't start foreground, I guess.

Also about this, it says **android.app.ForegroundServiceStartNotAllowedException: Service.startForeground() not allowed due to mAllowStartForeground false** My guess is that application is foreground and you can't start backgroundService while app is open. You may add some controls that if it is background startBackgroundService. If it is not, you just need to start your codes **without service.** Kind regards

**Comment by aykutuludag** [comment]
> id's approach to ensure user awareness of ongoin

Exacly, this is the expected behavior on notification's on mobile app. I will upgrade plugin on my project and released new version asap. Have a nice day.

**Comment by aykutuludag** [comment]
> > ![image](https://private-user-images.githubusercontent.com/18382503/298094320-512e36c6-0bcc-44c6-b59d-97a05fe98030.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU4MzczNDYsIm5iZiI6MTcwNTgzNzA0NiwicGF0aCI6Ii8xODM4MjUwMy8yOTgwOTQzMjAtNTEyZTM2YzYtMGJjYy00NGM2LWI1OWQtOTdhMDVmZTk4MDMwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTIxVDExMzcyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRiYjJlZmYzNWUyMzAxYjgxOTc4ODU4OTMxZDUzZDkyYzhmODQ5ODE0NWRhZjQ5ZTUwMzA1MzIzYWNkODdkY2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4MhH5TcGAzViBd5kEKq2sND1GMPCoyAUFFZmt_2_szs)
> > ```
> > Fatal Exception: java.lang.RuntimeException: Unable to start service com.gdelataillade.alarm.alarm.AlarmService@c777b30 with Intent { cmp=com.appname/com.gdelataillade.alarm.alarm.AlarmService (has extras) }: android.app.ForegroundServiceStartNotAllowedException: Service.startForeground() not allowed due to mAllowStartForeground false: service com.appname/com.gdelataillade.alarm.alarm.AlarmService
> >        at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:4963)
> >        at android.app.ActivityThread.-$$Nest$mhandleServiceArgs()
> >        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2288)
> >        at android.os.Handler.dispatchMessage(Handler.java:106)
> >        at android.os.Looper.loopOnce(Looper.java:211)
> >        at android.os.Looper.loop(Looper.java:300)
> >        at android.app.ActivityThread.main(ActivityThread.java:8410)
> >        at java.lang.reflect.Method.invoke(Method.java)
> >        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:559)
> >        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:954)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Other than this error, I couldn't detect any error so far on Android. Best regards.
> > **Note**: My suspect that this is only occurring when app is open alarm is triggered at that moment. Service couldn't start foreground, I guess.
> 
> Also about this, it says **android.app.ForegroundServiceStartNotAllowedException: Service.startForeground() not allowed due to mAllowStartForeground false** My guess is that application is foreground and you can't start backgroundService while app is open. You may add some controls that if it is background startBackgroundService. If it is not, you just need to start your codes **without service.** Kind regards
> 
> This error is only occurring Android >= 12.
> 
> Additional readings: https://stackoverflow.com/a/70666991/4606368

https://github.com/gdelataillade/alarm/issues/142 and https://github.com/gdelataillade/alarm/issues/141 is related with this error.

**Comment by gdelataillade** [comment]
Hi @aykutuludag 

I just released version `3.0.7` with some native android fixes. Hope it will fix your issues !

**Comment by aykutuludag** [comment]
Huge thanks for your effort. I'm applying v3.0.7 and sending new release. I will share if any bug encounters. Best regards.
